demo_nodes_py/demo_nodes_py/parameters/async_param_client.py

```diff
@@ -0,0 +1,118 @@
+
+
+def main(args=None):
+    target_node_name = 'param_test_target'
+    context = rclpy.context.Context()
+    rclpy.init(context=context)
+    target_executor = SingleThreadedExecutor(context=context)
+    client_executor = SingleThreadedExecutor(context=context)
+    node = rclpy.create_node('async_param_client', context=context)
+    target_node = rclpy.create_node(target_node_name,
+                                    allow_undeclared_parameters=True,
+                                    context=context)
+    target_node.declare_parameters('', [
+        ('int_parameter', 1),
+        ('bool_parameter', False),
+        ('string_parameter', 'Hello World'),
+    ])
+    target_executor.add_node(target_node)
+    client_executor.add_node(node)
+    thread = threading.Thread(target=target_executor.spin)
+    thread.start()
+    client = AsyncParameterClient(node, 'param_test_target')
+    param_list = ['int_parameter', 'bool_parameter', 'string_parameter']
+
+    node.get_logger().info('Listing Parameters:')
+    future = client.list_parameters(param_list, 10)
+    client_executor.spin_until_future_complete(future)
+    list_parameters_result = future.result()
+    if list_parameters_result is not None:
+        for param_name in list_parameters_result.result.names:
+            node.get_logger().info(f'    - {param_name}')
+    else:
+        node.get_logger().error(f'Error listing parameters: {future.exception()}')
+
+    node.get_logger().info('Setting parameters:')
+    future = client.set_parameters([
+        Parameter('int_parameter', Parameter.Type.INTEGER, 10),
+        Parameter('string_parameter', Parameter.Type.STRING, 'Fee Fi Fo Fum'),
+    ])
+    client_executor.spin_until_future_complete(future)
+    set_parameters_result = future.result()
+    if set_parameters_result is not None:
+        for i, v in enumerate(set_parameters_result.results):
+            node.get_logger().info(f'    {param_list[i]}:')
+            node.get_logger().info(f'        successful: {v.successful}')
+    else:
+        node.get_logger().error(f'Error setting parameters: {future.exception()}')
+
+    node.get_logger().info('Getting parameters:')
+    future = client.get_parameters(param_list)
+    client_executor.spin_until_future_complete(future)
+    get_parameters_result = future.result()
+    if get_parameters_result is not None:
+        for i, v in enumerate(get_parameters_result.values):
+            node.get_logger().info(f'    - {param_list[i]}: {parameter_value_to_python(v)}')
+    else:
+        node.get_logger().error(f'Error getting parameters: {future.exception()}')
+
+    node.get_logger().info('Loading parameters: ')
+    param_dir = get_package_share_directory('demo_nodes_py')
+    param_list = os.path.join(param_dir, 'params.yaml')
+    future = client.load_parameter_file(param_list)
+    client_executor.spin_until_future_complete(future)
+    load_parameter_results = future.result()
+    if load_parameter_results is not None:
```

Comment:
We're not doing anything here with `load_parameter_results`. Rather than re-reading the parameters file below, can we just use the load result?

Reply:
https://github.com/ros2/demos/blob/068ee28c0f5bcc1cb7900586417dce76bcbf238b/demo_nodes_py/demo_nodes_py/parameters/async_param_client.py#L144


Load result only tells you if its successful or not. I mistakenly removed it, thanks for the catch